pkg/agent/bgp/engine/gobgp.go

```diff
@@ -0,0 +1,295 @@
+// Copyright 2024 Antrea Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package engine
+
+import (
+	"context"
+	"fmt"
+	"net/netip"
+	"time"
+
+	gobgpapi "github.com/osrg/gobgp/v3/api"
+	"github.com/osrg/gobgp/v3/pkg/server"
+	"google.golang.org/protobuf/types/known/anypb"
+	"k8s.io/utils/net"
+)
+
+const (
+	ipv4AllZero = "0.0.0.0"
+	ipv6AllZero = "::"
+)
+
+type GoBGPServer struct {
+	server       *server.BgpServer
+	globalConfig *gobgpapi.Global
+}
+
+func NewGoBGPServer(globalConfig *GlobalConfig, extraLogFields map[string]interface{}) (Interface, error) {
+	logger, err := newGoBGPServerLogger(extraLogFields)
+	if err != nil {
+		return nil, err
+	}
+	return &GoBGPServer{
+		server: server.NewBgpServer(server.LoggerOption(logger)),
+		globalConfig: &gobgpapi.Global{
+			Asn:        globalConfig.ASN,
+			RouterId:   globalConfig.RouterID,
+			ListenPort: globalConfig.ListenPort,
+		},
+	}, nil
+}
+
+func (g *GoBGPServer) Start() error {
+	go g.server.Serve()
+	if err := g.server.StartBgp(context.TODO(), &gobgpapi.StartBgpRequest{Global: g.globalConfig}); err != nil {
+		return fmt.Errorf("failed to start BGP server: %w", err)
+	}
+	return nil
+}
+
+func (g *GoBGPServer) Stop() error {
+	if err := g.server.StopBgp(context.TODO(), &gobgpapi.StopBgpRequest{}); err != nil {
+		return fmt.Errorf("failed to stop BGP server: %w", err)
+	}
+	return nil
+}
+
+func (g *GoBGPServer) AddPeers(peerConfigs []PeerConfig, ipFamily net.IPFamily) error {
```

Comment:
> Why it doesn't just use BGPPeer in API as the argument? I think it would need to convert twice in the current code?

I think that can also work. However, `BGPPeer` in API has some pointers, and we need to complete them in these functions. Besides, when calling `UpdatePeers` (Maybe `UpdatePeer` later), we also need to complete them to verify that if there are some changes in `BGPPeer` except for `Address` and `ASN`.

> Doesn't address in PeerConfig already have family? Why it needs the ipFamily parameter?

If we refactor the method to `AddPeer`, we don't need the family. I added this because the method handles multiple BGP peers, and I think it is a little weird to get the family in the loop.